src/engine.ml

```diff
@@ -463,6 +463,9 @@ let maybe_push_reply config = function
         if String.starts_with str ~prefix:p_r then
           let opts = String.sub str p_r_len (String.length str - p_r_len) in
           Config.merge_push_reply config opts
+        else if String.starts_with str ~prefix:"AUTH_FAILED" then
+          let msg = List.tl (String.split_on_char ',' str) |> String.concat "," in
+          Error (`Msg msg)
```

Comment:
you test for "AUTH_FAILED" prefix, and then do a split on `,` and a List.tl -- would you mind to test for "AUTH_FAILED,"? (so we ensure there's at least two elements after the split?

Reply:
or otherwise do a `String.sub str 11 (String.length str - 11)` -- for AUTH_FAILED being 11 chars... (or use "AUTH_FAILED," and 12)...

Reply:
We probably can prepend the message with something like `authentication failed`, at least, the `List.tl` should never failed where `String.split_on_char` can not return an empty list. With the current code, the worst case will be to have an empty message.

Reply:
Done 👍